cmd/main.go

```diff
@@ -107,12 +108,18 @@ func main() {
 		Scheme: mgr.GetScheme(),
 	}
 
+	loggingSecrets := loggingcredentials.Reconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}
+
 	loggingReconciler := loggingreconciler.LoggingReconciler{
 		Client: mgr.GetClient(),
 		Scheme: mgr.GetScheme(),
 		Reconcilers: []reconciler.Interface{
 			&promtailReconciler,
 			&promtailWiring,
+			&loggingSecrets,
```

Comment:
What we do in other operators that we have is to run several reconcilers at the same time, instead of sequentially going one after the other. Have you consider using the other approach?

Reply:
We considered it, and thought given there's not much cluster creation/deletion we would not benefit from tasks parallelization, and it would mix logs and thus make debugging harder.

However I'm thinking maybe we should have several controllers so that we watch different resources independently. Like:
* cluster is updated, we update loggingCredentials secret
* secret is updated, we update grafana datasource

Or could it be done at reconciler level?

Reply:
Do you maybe have an example on how this in done in another operator Jose ?

Reply:
Yes, [we have two controllers here](https://github.com/giantswarm/aws-resolver-rules-operator/) and we are adding a third one in one of the open PRs.